### PR TITLE
Jhony/python 3 12

### PR DESCRIPTION
**Pull request by jfavellar90**
It aims to close https://github.com/overhangio/tutor-notes/issues/33

### CODE REVIEW

File: tutornotes/templates/notes/build/notes/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,21 @@
 # syntax=docker/dockerfile:1.4
-FROM docker.io/ubuntu:20.04
+FROM docker.io/python:3.12-slim-bookworm
```
Comment (DawoudSheraz) [inline tutornotes/templates/notes/build/notes/Dockerfile:2]: Any reason for choosing this explicit image?

File: tutornotes/templates/notes/build/notes/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-    python3-pip \
-    python3-venv \
-    libmysqlclient-dev \
-    pkg-config
-RUN ln -s /usr/bin/python3 /usr/bin/python
+    python3-dev \
+    build-essential \
+    default-libmysqlclient-dev \
+    pkg-config && \
+  sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && \
```
Comment (DawoudSheraz) [inline tutornotes/templates/notes/build/notes/Dockerfile:16]: why is this needed?

File: tutornotes/templates/notes/build/notes/Dockerfile (hunk 1 above)
Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile:2]: @DawoudSheraz I checked this [comment](https://github.com/overhangio/tutor-discovery/pull/67/files#r1533458049) and I thought it was a good opportunity to test with a Python image to avoid using Pyenv. I picked the slim one to get an image as light as possible.

File: tutornotes/templates/notes/build/notes/Dockerfile (hunk 2 above)
Comment (DawoudSheraz) [reply tutornotes/templates/notes/build/notes/Dockerfile:16]: Any links/references?

File: tutornotes/templates/notes/build/notes/Dockerfile (hunk 1 above)
Comment (DawoudSheraz) [reply tutornotes/templates/notes/build/notes/Dockerfile:2]: Alright. If it builds and works as expected, that is good.

File: tutornotes/templates/notes/build/notes/Dockerfile (hunk 2 above)
Comment (jfavellar90) [reply tutornotes/templates/notes/build/notes/Dockerfile:16]: @DawoudSheraz please check the link [here](https://stackoverflow.com/a/28406007).

**Review by DawoudSheraz — COMMENTED** [review]
Overall, looks good. I will try to verify the image build by tomorrow and give 👍🏽 on PR. Thanks

**Review by DawoudSheraz — APPROVED** [review]
Image is building as expected, notes feature appears to be working.
